tests/test_pure_pursuit.py

```diff
@@ -1,17 +1,86 @@
-import utilities.pure_pursuit as pp
+from utilities.pure_pursuit import PurePursuit
+from hypothesis.strategies import floats
+from hypothesis import given, assume
+import math
 
 
-def test_trapezoidal():
-    waypoints = [(0, 0, 0, 0), (10, 10, 0, 2)]
-    trap = pp.insert_trapezoidal_waypoints(waypoints, 2, -2)
-    assert len(trap) == 3
-    assert trap[1][2] == waypoints[1][2], (
-        "Intermediate waypoint should have end speed when accelerating: %s" % trap
-    )
+# def test_trapezoidal():
+#     waypoints = [(0, 0, 0, 0, 0), (10, 10, 0, 2, 0)]
+
+#     pp = PurePursuit(look_ahead=0.2,look_ahead_speed_modifier=0.25)
+#     trap = pp.insert_trapezoidal_waypoints(waypoints, 2, -2)
+#     assert len(trap) == 3
+#     assert trap[1][2] == waypoints[1][2], (
+#         "Intermediate waypoint should have end speed when accelerating: %s" % trap
+#     )
+
+#     waypoints = [(10, 10, 2, 0), (0, 0, 0, 0)]
+#     trap = pp.insert_trapezoidal_waypoints(waypoints, 2, -2)
+#     assert len(trap) == 3
+#     assert trap[1][2] == waypoints[0][2], (
+#         "Intermediate waypoint should have beginning speed when decelerating: %s" % trap
+#     )
+# for trapezoid function
 
-    waypoints = [(10, 10, 0, 2), (0, 0, 0, 0)]
-    trap = pp.insert_trapezoidal_waypoints(waypoints, 2, -2)
-    assert len(trap) == 3
-    assert trap[1][2] == waypoints[0][2], (
-        "Intermediate waypoint should have beginning speed when decelerating: %s" % trap
+
+@given(
+    floats(min_value=0, max_value=2),
+    floats(min_value=0, max_value=5),
+    floats(min_value=0, max_value=5),
+)
+def test_find_speed(dist, start_spd, end_spd):
+    pp = PurePursuit(look_ahead=0.2, look_ahead_speed_modifier=0.25)
+    start_path_distance = 0
+    end_path_distance = 2
+    start_speed = start_spd
+    end_speed = end_spd
+    distance_along_path = dist
```

Comment:
Why are you aliasing these? Just make the argument names the actual names you want.